R/cut.R

```diff
@@ -0,0 +1,105 @@
+#' Cut a numeric Crunch variable
+#' 
+#' crunch::cut() is equivalent to base::cut() except that it operates on 
+#' crunch variables instead of in-memory variables. The function divides the range of 
+#' x into intervals and codes the values in x according to which interval they fall. 
+#' The leftmost interval corresponds to level one, the next leftmost to level two and 
+#' so on.
+#' @param x A crunch variable of the class NumericVariable
+#' @param breaks Either a numeric vector of two or more unique cut points 
+#' or a single number (greater than or equal to 2) giving the number of intervals 
+#' into which x is to be cut.
+#' @param variable.name The name of the resulting case variable as a character string. 
+#' @param labels labels for the levels of the resulting category.
+#' By default, labels are constructed using interval notation.
+#' If labels = FALSE, simple integer codes are returned instead of a factor.
+#' @param include.lowest logical, indicating if an `x[i]` equal to the lowest 
+#' (or highest, for right = FALSE) `breaks` value should be included.
+#' @param right logical, indicating if the intervals should be closed on the right 
+#' (and open on the left) or vice versa.
+#' @param dig.lab	integer which is used when labels are not given. 
+#' It determines the number of digits used in formatting the break numbers.
+#' @param ordered_result	Ignored.
+#' @param ... further arguments passed to or from other methods.
+#'
+#'
+#' @return a Crunch VariableDefinition
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' ds <- loadDataset("mtcars")
+#' ds$cat_var <- cut(ds$mpg, 3, variableName = "new_var")
+#' }
+#'  
+setMethod("cut", "NumericVariable", function(x, 
+                                             breaks, 
+                                             variable.name, 
+                                             labels = NULL,
+                                             include.lowest = FALSE,
+                                             right = TRUE,
+                                             dig.lab = 3,
+                                             ordered_result = FALSE, ...){
+  env <- environment()
+  if (length(breaks) == 1L) {
+    if (is.na(breaks) || breaks < 2L) {
+      halt("invalid number of intervals")
+    }
+    nb <- as.integer(breaks + 1) # one more than #{intervals}
+    dx <- diff(rx <- c(min(x, na.rm = TRUE), max(x, na.rm = TRUE)))
+    if (dx == 0) {
+      dx <- abs(rx[1L])
+      breaks <- seq.int(rx[1L] - dx/1000, rx[2L] + dx/1000,
```

Comment:
For an integer "breaks" value, you could:

1) keep this copy-paste from `base::cut`;
2) reject the request (the main use for the function now is to specify specific cut points); or
3) return a VariableDefinition with a "bin" derivation function. The backend supports this autobinning already.

For expediency, I'd personally go with (2) and wait for someone to for it.